### DISCUSSION TITLE
Is it possible to enable copilot by filetype in vim/neovim?

### DISCUSSION THREAD

**Discussion by hukaidong** [community/community#60061] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I don't want copilot engage in some of my custom configuration file while using vim. Is it possible do disable copilot per buffer or per filetype?

**Comment by Akash1134** [comment]
Hi @hukaidong ,

Please check out the suggestions offered in this [similar discussion](https://github.com/orgs/community/discussions/11254). Hope that helps. 

Thanks!

**Reply by hukaidong** [reply]
Hi Akash:

I've seen those discussions. They are mainly focusing on GUI level but not yet support terminal operations. I think they haven't implement this feature yet.

Best

**Reply by hukaidong** [reply]
Also, the command ":Copilot disable" in vim disables copilot globally but not per buffer. This makes defining auto enabler/disabler impossible.

**Comment by rphuber** [comment, marked as answer]
there is a setting to allow you to disable copilot for a specific filetype.  if you are using vim to configure:
`let g:copilot_filetypes = {
    \ 'python': v:false,
    \ }`

you can also disable copilot for the current buffer with
`b:copilot_enabled` and setting that to v:false.

**Reply by michie1** [reply]
Small addition that solved my problem, you need to use the filetype and not the extension. You can retrieve the filetype of a file by running: `echo &filetype` when the specific file is opened.
